assemble/Zoltan_integration.F90

```diff
@@ -2073,17 +2072,10 @@ subroutine update_particle_list_element(detector_list_array)
        detector_list => detector_list_array(j)%ptr
        ewrite(2,*) "Length of detector list to be updated: ", detector_list%length
 
+       if (sum(detector_list%total_attributes)==0) return
```

Comment:
what's going on here? Can we just ignore any other lists in detector_list_array? This also assumes that `detector_list%total_attributes` is the same on all processes - which seems to contradict what's happening further below in the new bcasts...

Reply:
Good catch. This should cycle to the next detector list if `sum(detector_list%total_attributes)==0`. 
`detector_list%total_attributes` will be the same on all processes, I'll fix up the relevant contradictions below.